lib/commanded/event/handler.ex

```diff
@@ -821,14 +821,15 @@ defmodule Commanded.Event.Handler do
   end
 
   @doc false
+  @impl GenServer
+  def handle_info({:EXIT, _pid, :normal}, state) do
+    # linked process exited normally, don't shutdown
+    {:noreply, state}
+  end
```

Comment:
What would be the implications when it is gracefully shutting down?

You mention "don't shutdown," but if it is gracefully shutting down, I assume you should stop the process, no? How that work here?

Reply:
If it's gracefully shutting down, meaning being shut down by it's parent (supervisor) process, then the `terminate` callback is called, and no signal is sent. These signals are only for linked processes other than the parent, and for those we don't want the handler to shutdown when they exit.

The background story of this: https://github.com/bitcrowd/chromic_pdf/issues/233, in short: I was generating a pdf inside an event handler and in tests it runs in on demand mode, which means it starts and stops a supervisor/genserver just for that one request for generating a pdf. This shut down the event handler and made the tests fail in some situations.

Reply:
I believe this is a good change :+1: 

Background:
By default, the process is not trapping exits and then linking works like this for a link between A and B:
a) If process A dies with reason `:normal`, it means it finished itself, and everything is fine, so BEAM does NOT kill B.
b) If process A dies with any other reason, BEAM kills process B with the same reason.

However, `Commanded.Event.Handler` needs to trap exits to clean up in the `terminate` callback.
Beam then translates exit signals into messages.
1. If the message comes from the parent (the supervisor), the `terminate` callback is called.
2. Otherwise, the message comes from another linked process (e.g. someone `spawn_link`ed a process from within the handler.
Without this PR, all those exit signals would be logged as errors in the next `handle_info` clause (line 837) and worse, if a linked process died, the handler would not crash as expected.

So this PR brings back the sane default behaviour from a regular process that does not trap exits:
- the `{:EXIT, _pid, :normal}` covers case a)
- the `{:EXIT, _pid, reason}` covers case b)